### PR TITLE
Implement OpenAPI UI with static page approach

### PR DESCRIPTION
**Pull request by suho**
## What happened 👀

- Added a new controller (OpenapiController) and its view to show the OpenAPI UI.
- Updated config/application.rb to support allow CORS.
- Updated config/routes.rb to show OpenAPI UI in with `{{base_url}}/openapi`.
- Added tests for `OpenapiController`.

## Insight 📝

In this pull request, I decided to use the static page approach as it can be easily added to any of our web templates. If we use Docker, we might need to deploy with another server (app/service) or port.

I decided to use https://github.com/stoplightio/elements instead of Swagger UI, it has a better UI/UX (PoW)

## Proof Of Work 📹

|Swagger UI|Elements UI|
|-|-|
|Screenshot 2023-06-09 at 17 19 14|Screenshot 2023-06-15 at 11 52 46|

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,10 +21,15 @@ env/stop:
 env/teardown:
 	./bin/envteardown.sh
 
+doc/generate:
+	yarn run build:docs
+
 dev:
+	make doc/generate
```
Comment (olivierobert) [inline Makefile:28]: Why add that to the `dev` command? Same below for `make test`? I'm worried about the additional processing time.

File: config/routes.rb [hunk 2]
```diff
@@ -36,4 +36,6 @@
       resource :users, only: :show, path: :me
     end
   end
+
+  get '/openapi', to: 'openapi#show'
```
Comment (olivierobert) [inline config/routes.rb:40]: How about moving it to the namespace `api`?

```ruby
namespace :api do
  namespace :v1 do
    #...
  end

  namespace :docs do
     get '/openapi', to: 'openapi#show'
  end
end
```

File: config/application.rb [hunk 3]
```diff
@@ -29,5 +29,13 @@ class Application < Rails::Application
 
     # Automatically generate the `translation.js` files
     config.middleware.use I18n::JS::Middleware
+
+    # Use Rack::Cors to allow cross-origin requests in OpenAPI Swagger UI.
+    config.middleware.insert_before 0, Rack::Cors do
+      allow do
+        origins '*'
+        resource '*', headers: :any, methods: :any
```
Comment (olivierobert) [inline config/application.rb:37]: Will not this be applied to all routes? Rails provide [route constraints](https://guides.rubyonrails.org/routing.html#advanced-constraints) to limit what applies to specific routes (see https://tadhao.medium.com/constraints-in-rails-router-58f389031463)

File: Makefile (hunk 1 above)
Comment (suho) [reply Makefile:28]: Good catch! I wanted to generate docs every time when running `make dev` (or `make test`). Regarding processing time, it seems we only need to run one time. Let me make the update.

File: config/application.rb (hunk 3 above)
Comment (suho) [reply config/application.rb:37]: Thanks for this comment! #TIL

After rechecking, I think I will remove this, as this was set up for localhost. Currently, we can call our production URL :D

Comment (suho) [reply config/application.rb]: Removed this config in https://github.com/nimblehq/nimble-survey-web/pull/89/commits/6391ef47b371a40433b500dd5c0775c31a735eca

File: config/routes.rb (hunk 2 above)
Comment (suho) [reply config/routes.rb]: Fixed in https://github.com/nimblehq/nimble-survey-web/pull/89/commits/6391ef47b371a40433b500dd5c0775c31a735eca

File: Makefile (hunk 1 above)
Comment (suho) [reply Makefile]: Fixed in https://github.com/nimblehq/nimble-survey-web/pull/89/commits/6391ef47b371a40433b500dd5c0775c31a735eca

**Review by olivierobert — APPROVED** [review]

File: spec/requests/api/docs/openapi_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe API::Docs::OpenapiController, type: :request do
+  describe 'GET#show' do
+    context 'given an request' do
```
Comment (hoangmirs) [inline spec/requests/api/docs/openapi_controller_spec.rb:7]:
```suggestion
    context 'given a request' do
```

File: app/controllers/api/docs/openapi_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module API
+  module Docs
+    class OpenapiController < ApplicationController
+      def show
+        render 'show'
```
Comment (hoangmirs) [inline app/controllers/api/docs/openapi_controller.rb:7]:
```suggestion
        render :show
```

**Review by hoangmirs — APPROVED** [review]

File: spec/requests/api/docs/openapi_controller_spec.rb (hunk 4 above)
Comment (suho) [reply spec/requests/api/docs/openapi_controller_spec.rb]: Fixed in https://github.com/nimblehq/nimble-survey-web/pull/89/commits/43a2d65fc23d5253012f31f0b2fcdc505e1b3511

File: app/controllers/api/docs/openapi_controller.rb (hunk 5 above)
Comment (suho) [reply app/controllers/api/docs/openapi_controller.rb]: Fixed in https://github.com/nimblehq/nimble-survey-web/pull/89/commits/43a2d65fc23d5253012f31f0b2fcdc505e1b3511

**Pull request merged**